### PR TITLE
Create contribution documentation for openebs/e2e

### LINKED ISSUE #567: Improve the section for test contributions on e2e/ansible documentation
## Is this a BUG REPORT or FEATURE REQUEST?
  FEATURE REQUEST

**What happened**:
We need to elaborate the section in the e2e/ansible/README.md for user contributions. Details should include : 

- What we have currently
- What we are going to include
- Potential areas for contributions
- Best practices for contributions

### PR DESCRIPTION
**Pull request by ksatchit**
**What this PR does / why we need it**:

- Creates contribution documentation for openebs/e2e

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #567 

**Special notes for your reviewer**:
Created a single file - inspired in principle from kubernetes/e2e. Will add more sections for when applicable.

### CODE REVIEW

**Comment by kmova** [comment]
Very useful and nicely written doc @ksatchit

File: contribute/e2e/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
+## INTRODUCTION TO OPENEBS E2E
+
+OpenEBS e2e is focused on workload simulation & application deployment on OpenEBS storage, predominantly in a
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: and

File: contribute/e2e/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
+## INTRODUCTION TO OPENEBS E2E
+
+OpenEBS e2e is focused on workload simulation & application deployment on OpenEBS storage, predominantly in a 
+kubernetes environment. In its current state, it includes application workflow tests,basic compliance and resiliency tests. 
+The scope of the said tests is expected to evolve over time to include performance and security tests as well. 
+The e2e is mostly written in ansible, i.e., as ansible playbooks with docker used for test images (these include standard
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: Remove The before e2e
i.e. = that is

File: contribute/e2e/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
+## INTRODUCTION TO OPENEBS E2E
+
+OpenEBS e2e is focused on workload simulation & application deployment on OpenEBS storage, predominantly in a 
+kubernetes environment. In its current state, it includes application workflow tests,basic compliance and resiliency tests. 
+The scope of the said tests is expected to evolve over time to include performance and security tests as well. 
+The e2e is mostly written in ansible, i.e., as ansible playbooks with docker used for test images (these include standard 
+application images from dockerhub as well as custom-built images stored in openebs/test-storage).Since the tests are performed
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: Space after full stop

File: contribute/e2e/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
+## INTRODUCTION TO OPENEBS E2E
+
+OpenEBS e2e is focused on workload simulation & application deployment on OpenEBS storage, predominantly in a 
+kubernetes environment. In its current state, it includes application workflow tests,basic compliance and resiliency tests. 
+The scope of the said tests is expected to evolve over time to include performance and security tests as well. 
+The e2e is mostly written in ansible, i.e., as ansible playbooks with docker used for test images (these include standard 
+application images from dockerhub as well as custom-built images stored in openebs/test-storage).Since the tests are performed 
+in a kubernetes environment, the test images are typically deployed as "pods" and "jobs" and therefore accompanied by the
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: kubernetes = Kubernetes
Therefore they are accompanied...

File: contribute/e2e/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
+## INTRODUCTION TO OPENEBS E2E
+
+OpenEBS e2e is focused on workload simulation & application deployment on OpenEBS storage, predominantly in a 
+kubernetes environment. In its current state, it includes application workflow tests,basic compliance and resiliency tests. 
+The scope of the said tests is expected to evolve over time to include performance and security tests as well. 
+The e2e is mostly written in ansible, i.e., as ansible playbooks with docker used for test images (these include standard 
+application images from dockerhub as well as custom-built images stored in openebs/test-storage).Since the tests are performed 
+in a kubernetes environment, the test images are typically deployed as "pods" and "jobs" and therefore accompanied by the 
+respective specification YAML files.
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: YAML specification files.

File: contribute/e2e/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+kubernetes environment. In its current state, it includes application workflow tests,basic compliance and resiliency tests. 
+The scope of the said tests is expected to evolve over time to include performance and security tests as well. 
+The e2e is mostly written in ansible, i.e., as ansible playbooks with docker used for test images (these include standard 
+application images from dockerhub as well as custom-built images stored in openebs/test-storage).Since the tests are performed 
+in a kubernetes environment, the test images are typically deployed as "pods" and "jobs" and therefore accompanied by the 
+respective specification YAML files.
+
+## BUILDING BLOCKS OF OPENEBS CI
+
+The OpenEBS Continuous Integration has two distinct parts :
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: Remove "The"
parts:

File: contribute/e2e/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+The e2e is mostly written in ansible, i.e., as ansible playbooks with docker used for test images (these include standard 
+application images from dockerhub as well as custom-built images stored in openebs/test-storage).Since the tests are performed 
+in a kubernetes environment, the test images are typically deployed as "pods" and "jobs" and therefore accompanied by the 
+respective specification YAML files.
+
+## BUILDING BLOCKS OF OPENEBS CI
+
+The OpenEBS Continuous Integration has two distinct parts : 
+
+- Travis is used to run unit tests, build code and push images to dockerhub (in case of repos where the image is the build artifact)
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: repos = repositories??
the image = image

File: contribute/e2e/CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+- Jenkins pulls these images, creates the kubernetes-based test environment and executes the e2e. It is also used to run tests upon commits that change application specs OR openebs kubernetes deployment specs.
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: kubernetes-based = Kubernetes-based
executes the e2e = executes e2e
upon commits = on commits
specs = specifications
OR openebs kubernetes deployment specs = or OpenEBS Kubernetes deployment specifications

File: contribute/e2e/CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+
+While the OpenEBS e2e can be run as a standalone suite by users, the priority is have them run upon commits to the core openebs github
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: While you can run OpenEBS as a standalone suite, priority is to run them on commits done on core OpenEBS github.

File: contribute/e2e/CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+repositories such as openebs/openebs, openebs/mayaserver, openebs/jiva. This part of the continuous integration is built around a dedicated Jenkins server, which upon detecting commits via a polling mechanism, launches VMs using Vagrant and runs the ansible playbooks to setup the environment and execute e2e tests. The openebs/e2e project focuses on this part of the CI.
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: Jenkins server. On detecting commits through a polling mechanism, the server launches VMs.
CI = Continuous Integration should be expanded and within brackets (CI) at the first usage of CI.

File: contribute/e2e/CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+
+The below schema illustrates the workflow
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: workflow.

File: contribute/e2e/CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+
+![OpenEBS CI Overview Diagram](https://github.com/ksatchit/openebs/blob/master/documentation/source/_static/OpenEBS_CI_Workflow.png)
+
+## CONTRIBUTING TO E2E & OPENEBS CI
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: & = and

File: contribute/e2e/CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+While the OpenEBS e2e can be run as a standalone suite by users, the priority is have them run upon commits to the core openebs github
+repositories such as openebs/openebs, openebs/mayaserver, openebs/jiva. This part of the continuous integration is built around a dedicated Jenkins server, which upon detecting commits via a polling mechanism, launches VMs using Vagrant and runs the ansible playbooks to setup the environment and execute e2e tests. The openebs/e2e project focuses on this part of the CI.
+
+The below schema illustrates the workflow
+
+![OpenEBS CI Overview Diagram](https://github.com/ksatchit/openebs/blob/master/documentation/source/_static/OpenEBS_CI_Workflow.png)
+
+## CONTRIBUTING TO E2E & OPENEBS CI 
+
+Contributions are welcome !!  Before starting, a reading of the project documentation at [ReadTheDocs](http://openebs.readthedocs.io/en/latest/ ) is highly recommended. Please raise a github issue for any questions you may have on the project in general or e2e,CI in particular.
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: Before you start, CloudByte strongly recommends reading the project documentation available at ... it is highly....
in general or e2e, or CI in particular.

File: contribute/e2e/CONTRIBUTING.md [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+You can post questions on the OpenEBS slack channel too.
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: included slack link?

File: contribute/e2e/CONTRIBUTING.md [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+
+You could contribute to any of the following areas :
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: areas:

File: contribute/e2e/CONTRIBUTING.md [hunk 16]
```diff
@@ -0,0 +1,113 @@
...
+![OpenEBS CI Overview Diagram](https://github.com/ksatchit/openebs/blob/master/documentation/source/_static/OpenEBS_CI_Workflow.png)
+
+## CONTRIBUTING TO E2E & OPENEBS CI 
+
+Contributions are welcome !!  Before starting, a reading of the project documentation at [ReadTheDocs](http://openebs.readthedocs.io/en/latest/ ) is highly recommended. Please raise a github issue for any questions you may have on the project in general or e2e,CI in particular.
+You can post questions on the OpenEBS slack channel too.
+
+You could contribute to any of the following areas : 
+
+- Submit proposals to add new tests/test categories to the e2e
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: to e2e

Comment (mahebbar) [reply contribute/e2e/CONTRIBUTING.md]: tests/test is it a path? if not it should just be tests

File: contribute/e2e/CONTRIBUTING.md [hunk 17]
```diff
@@ -0,0 +1,113 @@
...
+- Create new application specification YAMLs 
+- Create ansible playbooks to execute test workflows
+- Submit changes to fix bugs/improve e2e tests
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: fix bugs or improve...

File: contribute/e2e/CONTRIBUTING.md [hunk 18]
```diff
@@ -0,0 +1,113 @@
...
+You can post questions on the OpenEBS slack channel too.
+
+You could contribute to any of the following areas : 
+
+- Submit proposals to add new tests/test categories to the e2e
+- Create new application specification YAMLs 
+- Create ansible playbooks to execute test workflows
+- Submit changes to fix bugs/improve e2e tests 
+- Raise/fix issues on the CI framework 
+- Raise/fix issues on the e2e, CI documentation
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: on e2e and CI documentation

File: contribute/e2e/CONTRIBUTING.md [hunk 19]
```diff
@@ -0,0 +1,113 @@
...
+
+## GENERAL CONTRIBUTION GUIDELINES
+
+Here are some general contribution guidelines. 
+
+### Adding new tests/test-categories
+
+To add new tests, please raise an issue describing the objective, pre-requisites, test steps, expected results & cleanup routines.
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: & = and

File: contribute/e2e/CONTRIBUTING.md [hunk 20]
```diff
@@ -0,0 +1,113 @@
...
+In case of proposals to add new test sections/categories, also add a generic description explaining the purpose and value addition
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: sections or categories, <remove also> add a generic....addition.

File: contribute/e2e/CONTRIBUTING.md [hunk 21]
```diff
@@ -0,0 +1,113 @@
...
+
+Once accepted, the issue can be assigned to interested contributors who can make relevant changes with a pull request. 
+
+### Create new application YAMLs
+
+To add new applications to e2e, please consider the points made here, as applicable :
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: please consider the following points, if applicable:

File: contribute/e2e/CONTRIBUTING.md [hunk 22]
```diff
@@ -0,0 +1,113 @@
...
+
+### Create new application YAMLs
+
+To add new applications to e2e, please consider the points made here, as applicable : 
+
+- https://kubernetes.io/docs/concepts/configuration/overview/
+- https://www.mirantis.com/blog/introduction-to-yaml-creating-a-kubernetes-deployment/
+- https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/
+
+These can be placed in here : https://github.com/openebs/openebs/tree/master/k8s/demo
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: remove in before here:

File: contribute/e2e/CONTRIBUTING.md [hunk 23]
```diff
@@ -0,0 +1,113 @@
...
+
+### Create ansible playbooks to execute test workflows
+
+The test playbooks can be placed in e2e/ansible/playbooks/<test-category>. Here are some considerations
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: considerations.

File: contribute/e2e/CONTRIBUTING.md [hunk 24]
```diff
@@ -0,0 +1,113 @@
...
+
+These can be placed in here : https://github.com/openebs/openebs/tree/master/k8s/demo
+
+### Create ansible playbooks to execute test workflows
+
+The test playbooks can be placed in e2e/ansible/playbooks/<test-category>. Here are some considerations 
+
+- Each test playbook is recommended to be constructed with the following template.
+
+  The main test playbook, which statically "includes" following files :
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: remove "which" statically...

File: contribute/e2e/CONTRIBUTING.md [hunk 25]
```diff
@@ -0,0 +1,113 @@
...
+
+  - Test variables YAML
+  - Pre-requisites tasks YAML
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: Pre-requisite??

File: contribute/e2e/CONTRIBUTING.md [hunk 26]
```diff
@@ -0,0 +1,113 @@
...
+  - Cleanup tasks YAML
+  
+  Auxiliary scripts that are invoked in the test playbook can be placed in the same directory. 
+
+  A sample test playbook can be seen [here](https://github.com/openebs/openebs/tree/master/e2e/ansible/playbooks/hyperconverged/test-k8s-percona-mysql-pod)
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: You can view a sample test playbook [here].

File: contribute/e2e/CONTRIBUTING.md [hunk 27]
```diff
@@ -0,0 +1,113 @@
...
+
+- Test playbooks that deploy applications are recommended to include steps to generate application-specfic workloads that
+  will exercise the OpenEBS storage volume. 
+  
+- Here are some ansible best practices while constructing playbooks : 
+ 
+  - https://www.jeffgeerling.com/blog/yaml-best-practices-ansible-playbooks-tasks 
+  
+  The OpenEBS ansible environment setup is detailed [here](https://github.com/openebs/openebs/blob/master/e2e/ansible/openebs-on-premise-deployment-guide.md)
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: setup details are available [here]

File: contribute/e2e/CONTRIBUTING.md [hunk 28]
```diff
@@ -0,0 +1,113 @@
...
+  
+### Submit changes to fix bugs/improve e2e tests
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: bugs or improve

File: contribute/e2e/CONTRIBUTING.md [hunk 29]
```diff
@@ -0,0 +1,113 @@
...
+  
+- Here are some ansible best practices while constructing playbooks : 
+ 
+  - https://www.jeffgeerling.com/blog/yaml-best-practices-ansible-playbooks-tasks 
+  
+  The OpenEBS ansible environment setup is detailed [here](https://github.com/openebs/openebs/blob/master/e2e/ansible/openebs-on-premise-deployment-guide.md)
+  
+### Submit changes to fix bugs/improve e2e tests 
+
+Please explain the fix made in detail and add the verbose output of the test playbook run with the fixes (Run the playbook with a -v option for this). Also, do consider the general ansible playbook best practices referenced in the previous section.
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: Explain the fix in detail and add the verbose test playbook output that is run with the fixes (run...
Also, consider the...

File: contribute/e2e/CONTRIBUTING.md [hunk 30]
```diff
@@ -0,0 +1,113 @@
...
+
+### Raise/fix issues on the CI framework
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: Raise or fix

File: contribute/e2e/CONTRIBUTING.md [hunk 31]
```diff
@@ -0,0 +1,113 @@
...
+  
+  The OpenEBS ansible environment setup is detailed [here](https://github.com/openebs/openebs/blob/master/e2e/ansible/openebs-on-premise-deployment-guide.md)
+  
+### Submit changes to fix bugs/improve e2e tests 
+
+Please explain the fix made in detail and add the verbose output of the test playbook run with the fixes (Run the playbook with a -v option for this). Also, do consider the general ansible playbook best practices referenced in the previous section.
+
+### Raise/fix issues on the CI framework 
+
+As with other issues, please mention the bug/enhancements in detail with logs/references if possible, respectively.
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: As you have done with other issues, ...  
remove respectively??

File: contribute/e2e/CONTRIBUTING.md [hunk 32]
```diff
@@ -0,0 +1,113 @@
...
+  
+### Submit changes to fix bugs/improve e2e tests 
+
+Please explain the fix made in detail and add the verbose output of the test playbook run with the fixes (Run the playbook with a -v option for this). Also, do consider the general ansible playbook best practices referenced in the previous section.
+
+### Raise/fix issues on the CI framework 
+
+As with other issues, please mention the bug/enhancements in detail with logs/references if possible, respectively. 
+
+The jenkins configuration used to setup the OpenEBS e2e can be found [here](https://github.com/openebs/openebs/blob/master/e2e/jenkins/README.md)
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: setup OpenEBS

File: contribute/e2e/CONTRIBUTING.md [hunk 33]
```diff
@@ -0,0 +1,113 @@
...
+
+### Raise/fix issues on the e2e, CI documentation
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: Raise or fix .... on e2e

File: contribute/e2e/CONTRIBUTING.md [hunk 34]
```diff
@@ -0,0 +1,113 @@
...
+
+Getting Documentation Right is Hard! Please raise a PR with you proposed changes to the READMEs, tutorials, deployment guides etc.,
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: etc. = and so on.

File: contribute/e2e/CONTRIBUTING.md (hunk 1 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 2 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 3 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 4 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 5 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 6 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 7 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 8 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 9 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 10 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 11 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 12 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 13 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Not sure about adding CB here. Made the other change

File: contribute/e2e/CONTRIBUTING.md (hunk 14 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: yes

File: contribute/e2e/CONTRIBUTING.md (hunk 15 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 16 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: retained just tests

File: contribute/e2e/CONTRIBUTING.md (hunk 17 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 18 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 19 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 20 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 21 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 22 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 23 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 24 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 25 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 26 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 27 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 28 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 29 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 30 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 31 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Not really sure if all contributors already have their "own" issues. Retained "As with other issues". Removed the "respectively"

File: contribute/e2e/CONTRIBUTING.md (hunk 32 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 33 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: Done

File: contribute/e2e/CONTRIBUTING.md (hunk 34 above)
Comment (ksatchit) [reply contribute/e2e/CONTRIBUTING.md]: restricted to these 3 - readme, tutorial and deployment guides.

File: contribute/e2e/CONTRIBUTING.md [hunk 35]
```diff
@@ -0,0 +1,112 @@
+## INTRODUCTION TO OPENEBS E2E
+
+OpenEBS e2e is focused on workload simulation and application deployment on OpenEBS storage, predominantly in a 
+kubernetes environment.In its current state, it includes application workflow tests, basic compliance and resiliency tests.
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: Kubernetes
tests, <space?

File: contribute/e2e/CONTRIBUTING.md [hunk 36]
```diff
@@ -0,0 +1,112 @@
...
+  - Cleanup tasks YAML
+  
+  Auxiliary scripts that are invoked in the test playbook can be placed in the same directory. 
+
+ You can view a sample test playbook [here](https://github.com/openebs/openebs/tree/master/e2e/ansible/playbooks/hyperconverged/test-k8s-percona-mysql-pod)
+
+- Test playbooks that deploy applications are recommended to include steps to generate application-specfic workloads that
+  will exercise the OpenEBS storage volume. 
+  
+- Here are some ansible best practices while constructing playbooks :
```
Comment (mahebbar) [inline contribute/e2e/CONTRIBUTING.md]: playbooks:

**Pull request merged**